lib/src/platform_client.dart

```diff
@@ -186,13 +191,52 @@ class PlatformClient {
       'context': jsonEncode(context),
       'requestSource': _config.clientId,
       'requestType': 'AIRA', // Required but unused.
+      'hasMessage': null != message && message.isNotEmpty,
+      'message': '$message${fileNames.isEmpty ? '' : ' (With files: $fileNames)'}',
+      'fileIds': fileIds,
+      'cannotTalk': cannotTalk ?? false,
       'useWebrtcRoom': true,
     });
 
     ServiceRequest serviceRequest =
         ServiceRequest.fromJson(await _httpPost('/api/user/$_userId/service-request', body));
 
-    return KurentoRoom.create(_config.environment, this, _session!, _pubnub, serviceRequest, roomHandler);
+    messagingClient?.serviceRequestId = serviceRequest.id;
+
+    return KurentoRoom.create(_config.environment, this, _session!, messagingClient, serviceRequest, roomHandler);
+  }
+
+  Future<List<String>> _sendPreCallMessage(String? text, Map<String, List<int>>? fileMap) async {
+    if (null == messagingClient) {
+      throw UnsupportedError('The application does not support messaging');
+    }
+    _log.finest('Sending pre-call message (message: $text, files: ${fileMap?.keys.join(', ')})');
+    String? message = text?.trim();
+    await messagingClient!.sendStart();
+    if (null != fileMap && fileMap.isNotEmpty) {
+      if (fileMap.length == 1) {
+        // If we have only one file, send it with the file.
+        var fileEntry = fileMap.entries.first;
+        SentFileInfo fileInfo = await messagingClient!.sendFile(fileEntry.key, fileEntry.value, text: message);
+        return [fileInfo.id];
+      } else {
+        // if we have multiple files, send them separately from teh message
+        if (null != message && message.isNotEmpty) {
+          // Waiting on first message separately to insure it gets to the server first.
+          await messagingClient!.sendMessage(message);
+        }
+
+        List<Future<SentFileInfo>> futureFileInfo = fileMap.entries.map((e) =>
+            messagingClient!.sendFile(e.key, e.value)).toList(growable: false);
+        List<SentFileInfo> fileInfoList = await Future.wait(futureFileInfo);
+
+        List<String> fileIds = fileInfoList.map((fi) => fi.id).toList(growable: false);
+        return fileIds;
+      }
+    } else if (null != text && text.isNotEmpty){
+      await messagingClient!.sendMessage(text);
+    }
```

Comment:
The idea of separating messages this way is to make it easier to manage later into the history. If we have many files, we should be able to delete any of those separately (hence separating them from the message). If we have only one file to send and it is accompanied by a file, I keep them together as the user will most likely want to delete them both if he ever wants to delete them.